### ISSUE TITLE
Add a lint rule specifically for ember-modifier modifiers to not destructure arguments, as they can lead to backtracking re-render assertions

### ISSUE THREAD

**Issue by NullVoxPopuli** [ember-cli/eslint-plugin-ember#2346]
We want tracked property access to be as lazy as possible, and typically when we're using modifiers that add event listeners, we don't care to consume that tracked data until much later -- like when the listened events fire -- so consuming the tracked data should be held off until then.

Example (bad):
```js
modifier((element, [text]) => {
  element.addEventListener('hover', () => console.log(text));
});
```
Example (good):
```js
modifier((element, positional) => {
  element.addEventListener('hover', () => console.log(positional[0]))
});
```
